torch/testing/__init__.py

```diff
@@ -250,6 +250,40 @@ def make_non_contiguous(tensor: torch.Tensor) -> torch.Tensor:
     # Use .data here to hide the view relation between input and other temporary Tensors
     return input.data
 
+# Common type groups
+_common_float_types = (torch.float32, torch.float64)
+_common_float_types_plus_half = _common_float_types + (torch.float16,)
+_common_float_types_plus_bfloat16 = _common_float_types + (torch.bfloat16,)
+_all_float_types = (torch.float16, torch.bfloat16, torch.float32, torch.float64)
+_complex_types = (torch.complex64, torch.complex128)
+_common_float_and_complex_types = _common_float_types + _complex_types
+_common_float_and_complex_types_plus_half = _common_float_types_plus_half + _complex_types
+_common_float_and_complex_types_plus_bfloat16 = _common_float_types_plus_bfloat16 + _complex_types
+_all_float_and_complex_types = _all_float_types + _complex_types
+
+def get_common_float_types():
```

Comment:
Some comments about why we have so many different variations of these functions would be helpful for future users.

Also, you've put this in `torch.testing` rather than `torch.testing._internal`, which implies that these functions are fair game for arbitrary third-party users to use. Was this intended?

Reply:
We do have too many variants! And I was sneakily trying to resolve that problem here, too. These particular lists are a good fit for the tests we write in practice. I can move them to _internal just in case. I put them here only because that's where existing dtype lists are.

Reply:
Well, I mean, in general, it's better not to publish as public API things you aren't sure you want to be public API. Easy to go private-public, hard to go other way.